### ISSUE TITLE
Migration: Empty tables from DVN 3.6.3 result in null ids in DV 4 tables

### ISSUE THREAD

**Issue by tdilauro** [IQSS/dataverse#3161]
During migration from DVN 3.6.3 to DV 4, I ran into mysterious referential integrity issues running migrate_users.sql in [Step 1 of the migration instructions](https://github.com/IQSS/dataverse/blob/develop/scripts/migration/migration_instructions.txt#L24). This was because the id column had been inadvertently set to null by the [offset adjustments SQL commands in the migration pre-steps](https://github.com/IQSS/dataverse/blob/develop/scripts/migration/migration_presteps.txt#L129-L136).

My understanding is that the goal is to increment the ids of various tables by the largest id of the DVN 3 vdcnetwork table. The problem results from using the query:
`SELECT MAX(id) FROM _dvn3_vdcnetwork`

This gives the desired result on a table with one or more rows, where it returns the maximum value for the id column; but for a  table with no rows, it returns null, rather than zero. When that null is added to any other value, the result is also null. This left all the UPDATE / INSERT target ids = null.

To achieve the desired result, the query should be:
`SELECT COALESCE(MAX(id), 0) FROM _dvn3_vdcnetwork`

The following patch corrected the issue for us:
```
$ git diff scripts/migration/migration_presteps.txt
diff --git a/scripts/migration/migration_presteps.txt b/scripts/migration/migration_presteps.txt
index 517a967..df2be49 100644
--- a/scripts/migration/migration_presteps.txt
+++ b/scripts/migration/migration_presteps.txt
@@ -127,13 +127,13 @@ psql -h localhost -U postgres <4.0 database name> -f /tmp/dvn3_data.sql
 ----------------------------------------------

 -- offsets
-update _dvn3_vdcnetwork set id = id + (select max(id) from dvobject);
-update _dvn3_vdc set id = id + (select max(id) from _dvn3_vdcnetwork);
-update _dvn3_vdcrole set vdc_id = vdc_id + (select max(id) from _dvn3_vdcnetwork);
-update _dvn3_vdc_usergroup set vdcs_id = vdcs_id + (select max(id) from _dvn3_vdcnetwork);
-update _dvn3_vdc_linked_collections set vdc_id = vdc_id + (select max(id) from _dvn3_vdcnetwork);
-update _dvn3_study set owner_id = owner_id + (select max(id) from _dvn3_vdcnetwork);
-update _dvn3_vdccollection set owner_id = owner_id + (select max(id) from _dvn3_vdcnetwork);
+update _dvn3_vdcnetwork set id = id + (select coalesce(max(id), 0) from dvobject);
+update _dvn3_vdc set id = id + (select coalesce(max(id), 0) from _dvn3_vdcnetwork);
+update _dvn3_vdcrole set vdc_id = vdc_id + (select coalesce(max(id), 0) from _dvn3_vdcnetwork);
+update _dvn3_vdc_usergroup set vdcs_id = vdcs_id + (select coalesce(max(id), 0) from _dvn3_vdcnetwork);
+update _dvn3_vdc_linked_collections set vdc_id = vdc_id + (select coalesce(max(id), 0) from _dvn3_vdcnetwork);
+update _dvn3_study set owner_id = owner_id + (select coalesce(max(id), 0) from _dvn3_vdcnetwork);
+update _dvn3_vdccollection set owner_id = owner_id + (select coalesce(max(id), 0) from _dvn3_vdcnetwork);

 -- note: need to determine what offset to use, based on the file scripts
 --update _dvn3_studyfile_vdcuser set studyfiles_id = studyfiles_id +100000;
```

I suspect that we hit this problem because we are not using all the features of the features of DVN.

**Comment by tdilauro** [comment]
https://github.com/tdilauro/dataverse/commit/1ca2e9b0e1b64c20e64717707e9225e7b24e73b8

**Comment by pdurbin** [comment]
@tdilauro fixed this in pull request #3162 which was commit 1ca2e9b in 4.5 (I'll add the milestone) so we should go ahead and close this issue. I'm moving it to QA at https://waffle.io/IQSS/dataverse
